packages/block-editor/src/components/block-navigation/block.js

```diff
@@ -45,6 +47,14 @@ export default function BlockNavigationBlock( {
 		'block-editor-block-navigation-block__mover-cell',
 		{ 'is-visible': hasVisibleMovers }
 	);
+	const {
+		__experimentalWithEllipsisMenu: withEllipsisMenu,
+		__experimentalWithEllipsisMenuMinLevel: ellipsisMenuMinLevel,
```

Comment:
Small thing, but I think we shouldn't use the term EllipsisMenu for the menu.

Elsewhere in the code it's called BlockSettings, so I think it'd be good to stay consistent with that.

Reply:
I have been pondering this exact naming. The two are not exactly the same though. BlockSettings offers certain set of options:

<img width="798" alt="Zrzut ekranu 2020-05-21 o 11 40 08" src="https://user-images.githubusercontent.com/205419/82546069-d79be680-9b57-11ea-8b73-d171c398cfe5.png">

And the ellipsis menu is supposed to include everything from BlockSettings PLUS everything from the toolbar as well:

<img width="500" alt="Zrzut ekranu 2020-05-21 o 11 40 47" src="https://user-images.githubusercontent.com/205419/82546158-f7cba580-9b57-11ea-8134-c2d857d4b3bd.png">

While I don't love the name EllipsisMenu, I am also not convinced that using the name BlockSettings for both menus is the right thing to do here. Let's discuss that more on https://github.com/WordPress/gutenberg/pull/22463 - we may need to rethink these names a little bit more as we also have BlockControls and then UniversalBlockControls which I believe is too many similar names in circulation :-)

Reply:
In either case `EllipsisMenu` is bad because it references the icon we use for the menu. What if we change that icon to a circle?

If we want a menu that includes the `BlockControls` and the `BlockSettings` which is used by a Block Navigator Item, we'd wrap the two (for reuse if that is the case) in a thing that references the functionality, for example `BlockNavigatorItemSettings`.

Also, since `BlockSettings` has Slots we can use these to extend that in the `BlockNavigatorItemSettings`. If we want slots to be available for actions/ items specific to to `BlockNavigatorItem` then we'd add slots to `BlockNavigatorItemSettings`.

Reply:
@draganescu as I mentioned I agree we shouldn't call it `EllipsisMenu` - I was simply looking for a better option :-) `BlockNavigatorItemSettings` sounds pretty good so I went with it in https://github.com/WordPress/gutenberg/pull/22463

Reply:
Ah, I don't think `BlockNavigatorItemSettings` is right either. The `Navigator` is called `Navigation` internally and the `Item` is called `Block`, so I reckon `BlockNavigationBlockSettings` is perhaps the right way to go.

I know it's weirdly repetitive, but it's just a component name, and it makes it pretty clear where it lives and that it's a version of the `BlockSettings`.

Reply:
@adamziel @draganescu @talldan I'm confused here, why shouldn't the two menus be the same?

Reply:
Because in the Navigator the Items might have additional options that don't make sense in the block's menu. I don't know if that is the case @mtias. One example is add submenu, but that should sit fine with the block as well, why not?

Reply:
Yes, they seem to serve conceptually the exact same purpose. If there are certain features that would not make sense on one or the other, we should probably handle it through configuration / context.